connectors/webhook/src/main/java/io/camunda/connector/inbound/signature/strategy/StrategyContext.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.inbound.signature.strategy;
+
+import io.camunda.connector.inbound.model.HMACScope;
+import io.camunda.connector.inbound.utils.HttpMethods;
+import java.util.Arrays;
+
+class StrategyContext {
+  private final HMACScope[] hmacScopes;
+  private final String method;
+
+  public StrategyContext(HMACScope[] hmacScopes, String method) {
+    this.hmacScopes = hmacScopes;
+    this.method = method;
+  }
+
+  public boolean containsAllScopes(HMACScope... scopes) {
+    return hmacScopes.length == scopes.length
+        && Arrays.stream(hmacScopes).allMatch(s -> Arrays.asList(scopes).contains(s));
+  }
+
+  public boolean isGetMethod() {
+    return HttpMethods.get.name().equalsIgnoreCase(method);
+  }
```

Comment:
Suggestion: we could place this functionality into static methods in `HMACEncodingStrategyFactory` and avoid creating objects of this class